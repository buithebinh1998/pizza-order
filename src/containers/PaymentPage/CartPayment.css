.paycart {
  color: red;
  background-color: white;
  height: 25rem;
  position: relative;
  border: #04354c solid 1px;
  border-radius: 1rem;
  margin-bottom: 1rem;
}

.paycart-title {
  margin-top: 1rem;
  font-size: 1.2rem;
  font-weight: bold;
}

.pay-total-wrap {
  padding: 0 0.5rem;
  display: flex;
  font-size: 1.2rem;
  justify-content: space-between;
  position: absolute;
  bottom: 0.8rem;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
}

.paycart-price {
  font-weight: bold;
}

.paycart-info {
  text-align: left;
  margin-top: 1rem;
  color: #343434;
  font-size: 1.2rem;
  font-weight: bold;
  margin-left: 0.2rem;
  margin-right: 0.2rem;
}

.paycart-info-name {
  font-weight: bold;
  text-align: left;
  padding-left: 0.2rem;
}

.paycart-info-price-wrapper {
  display: flex;
  margin-top: 0.5rem;
  padding-left: 0.2rem;
  padding-right: 0.5rem;
}

.paycart-info-price {
  margin-left: auto;
}

.paycart-load {
  height: 20rem;
  overflow-y: scroll;
}

.paycart-remove-button {
  font-size: 1rem;
  margin-top: 0.5rem;
}

@media (max-width: 992px) {
  .paycart {
    border-radius: 0.8rem;
    margin-bottom: 1rem;
  }

  .paycart-title {
    margin-top: 0.8rem;
    margin-bottom: 0.2rem;
    font-size: 1rem;
  }

  .pay-total-wrap {
    padding: 0 0.5rem;
    font-size: 1rem;
    bottom: 3rem;
  }

  .paycart-checkout {
    font-size: 1rem;
    width: 8rem;
    height: 2rem;
    border-radius: 0.2rem;
    bottom: 0.5rem;
  }

  .paycart-info {
    margin-top: 0.8rem;
    font-size: 1rem;
    margin-left: 0.2rem;
    margin-right: 0.2rem;
  }

  .paycart-info-price-wrapper {
    margin-top: 0.5rem;
    padding-left: 0.2rem;
    padding-right: 0.5rem;
  }

  .paycart-info-quantity {
    margin: 0 0.2rem;
  }

  .paycart-remove-button {
    font-size: 0.8rem;
    margin-top: 0.5rem;
  }
}

@media (max-width: 768px) {
  .paycart {
    height: 20rem;
  }

  .paycart-load {
    height: 15rem;
  }

  .pay-total-wrap {
    bottom: 0.5rem;
  }
}

@media (max-width: 414px) {
  .paycart-title {
    font-size: 0.8rem;
  }

  .paycart {
    height: 18rem;
  }

  .paycart-load {
    height: 12rem;
  }

  .pay-total-wrap {
    font-size: 0.8rem;
  }
  .paycart-info {
    font-size: 0.8rem;
    margin-right: 0.2rem;
  }
  .paycart-remove-button {
    font-size: 0.6rem;
  }
}

.swal-title {
  color: #343434;
}
