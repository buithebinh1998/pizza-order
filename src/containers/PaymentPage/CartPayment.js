import React, {useContext, useEffect} from 'react'
import './CartPayment.css'
import {Context} from '../../context/Context/Context'
import {withRouter} from 'react-router-dom'
import swal from 'sweetalert'
const CartPayment = (props) => {
    const {cart, setNewTotalPrice, removeFromCart} = useContext(Context);
    let totalPrice = 0;
    let history = props.history;
    const checkCartPayment = () => {
        if(cart.length===1) {
            swal({
                title: "Your cart is empty!",
                text: 'Redirecting to Category page!',
                icon: 'warning',
                timer: 3000
            })
            const newCart = []; 
            localStorage.setItem("cart", JSON.stringify(newCart));
            setTimeout(() => {history.push('/category')}, 3000);
        }
    }
    
    const loadCart = cart.map((item, index) => {
        const showPrice = item.price*1000*item.quantity;
        const showPriceString = new Intl.NumberFormat('de-DE').format(showPrice); 
        totalPrice+=showPrice;
        setNewTotalPrice(totalPrice);
        
        return(
            <div className="paycart-info" key={index}>
                <div className="paycart-info-name">{item.name}</div>
                <div className="paycart-info-price-wrapper">
                    
                    <div className="paycart-info-quantity">x{item.quantity}</div>
                    
                    <div className="paycart-info-price">{showPriceString+"Đ"}</div>
                </div>
                <button className="paycart-remove-button" onClick={() => {removeFromCart(item); checkCartPayment();}}>Remove</button>
            </div>
        )
    });

    useEffect( () => {
        localStorage.setItem("cart", JSON.stringify(cart));
    }, [cart]);

    return(
        <div className="paycart">
            <div>
                <div className="paycart-title">ORDER INFORMATION</div>
                <div className="paycart-load">
                    {loadCart}
                </div>
                <div className="pay-total-wrap">
                    <div className="paycart-total">Total:</div>
                    <div className="paycart-price">{Intl.NumberFormat('de-DE').format(totalPrice)+"Đ"}</div>
                </div>
            </div>
            
        </div>
    )
}

export default withRouter(CartPayment);