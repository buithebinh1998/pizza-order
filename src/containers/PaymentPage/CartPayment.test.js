import { testSnapshots } from "../../utils/test.util";

import CartPayment from "./CartPayment";

jest.mock('react-router-dom', () => ({
    withRouter: (object) => object
}))
describe("<CartPayment/ >", () => {
  testSnapshots(CartPayment, [
    {
      props: {
          history: {
              push: jest.fn()
          }
      },
      description: "default render"
    }
  ]);
});
