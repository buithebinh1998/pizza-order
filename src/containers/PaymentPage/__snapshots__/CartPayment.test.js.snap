// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartPayment/ > snapshots default render 1`] = `
<div
  className="paycart"
>
  <div>
    <div
      className="paycart-title"
    >
      ORDER INFORMATION
    </div>
    <div
      className="paycart-load"
    />
    <div
      className="pay-total-wrap"
    >
      <div
        className="paycart-total"
      >
        Total:
      </div>
      <div
        className="paycart-price"
      >
        0ƒê
      </div>
    </div>
  </div>
</div>
`;
